<launch>

    <arg name="camera_name" default="stereo" />
    <arg name="device_name" default="/dev/video1" />
    <arg name="fps" default="30" />
    <arg name="frame_id" default="$(arg camera_name)" />
    <arg name="camera_info_url" default="" />
    <arg name="width" default="744" />
    <arg name="height" default="480" />
    <arg name="image_format" default="bayer_gbrg8" />
    <arg name="show" default="true" />

    <group ns="$(arg camera_name)">

        <group ns="left">
            <node pkg="lirs_video_stream" type="video_stream" name="$(arg camera_name)_video_stream" output="screen">
                <param name="camera_name" type="string" value="$(arg camera_name)" />
                <param name="device_name" type="string" value="/dev/v4l/by-id/usb-The_Imaging_Source_Europe_GmbH_DFM_22BUC03-ML_03610446-video-index0" />
                <param name="fps" type="int" value="$(arg fps)" />
                <param name="frame_id" type="string" value="$(arg frame_id)" />
                <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
                <param name="width" type="int" value="$(arg width)" />
                <param name="height" type="int" value="$(arg height)" />
                <param name="image_format" type="string" value="$(arg image_format)" />
                 <remap from="image" to="image_raw" />
            </node>

            <node if="$(arg show)" name="image_view" pkg="image_view" type="image_view">
                <remap from="image" to="image_raw" />
            </node>

        </group>

        <group ns="right">
            <node pkg="lirs_video_stream" type="video_stream" name="$(arg camera_name)_video_stream" output="screen">
                <param name="camera_name" type="string" value="$(arg camera_name)" />
                <param name="device_name" type="string" value="/dev/v4l/by-id/usb-The_Imaging_Source_Europe_GmbH_DFM_22BUC03-ML_03610453-video-index0" />
                <param name="fps" type="int" value="$(arg fps)" />
                <param name="frame_id" type="string" value="$(arg frame_id)" />
                <param name="camera_info_url" type="string" value="$(arg camera_info_url)" />
                <param name="width" type="int" value="$(arg width)" />
                <param name="height" type="int" value="$(arg height)" />
                <param name="image_format" type="string" value="$(arg image_format)" />
                <remap from="image" to="image_raw" />
            </node>

            <node if="$(arg show)" name="image_view" pkg="image_view" type="image_view">
                <remap from="image" to="image_raw" />
            </node>

        </group>

    </group>

</launch>
